### DISCUSSION TITLE
Shared object files or dynamic libraries

### DISCUSSION THREAD

**Discussion by rfcard** [zephyrproject-rtos/zephyr#68782] [Q&A] [answered]
I would like to be able to create dynamic libraries (".so" shared object files) that my Zephyr application can download when it needs functionality.

Is it possible to do this today, but in another way?

**Comment by MaikVermeulen** [comment]
I know this is old, but I'm very interested in this.

**Comment by carlescufi** [comment]
We would need a bit more information as to what you would like to do.
If you want examples on how to distribute code as shared libraries (`.a`) you can take a look at the [sdk-nrfxlib](https://github.com/nrfconnect/sdk-nrfxlib) repository of the Nordic nRF Connect SDK, which is pulled by the main [sdk-nrf manifest](https://github.com/nrfconnect/sdk-nrf/blob/master/west.yml)

**Comment by carlescufi** [comment, marked as answer]
Note that Zephyr does not support dynamic linking at all, so the `.a` files need to be present at build time

**Comment by nashif** [comment]
also look at LLEXT, might be what you need.
